opencsp/common/lib/csp/StandardPlotOutput.py

```diff
@@ -0,0 +1,556 @@
+            self.fig_control, self.axis_control, name='Ray Trace Image ' + suffix, number_in_name=False
+        )
+        fig_rec.axis.imshow(
+            ray_trace_data.histogram,
+            cmap='jet',
+            extent=(
+                ray_trace_data.histogram_x.min(),
+                ray_trace_data.histogram_x.max(),
+                ray_trace_data.histogram_y.min(),
+                ray_trace_data.histogram_y.max(),
+            ),
+        )
+        if self._to_save:
+            fig_rec.save(
+                output_dir=self.options_file_output.output_dir,
+                dpi=self.options_file_output.save_dpi,
+                format='png',
+                close_after_save=self.options_file_output.close_after_save,
+            )
+
+    def _perform_ray_trace_optic_measured(self):
+        """Performs ray trace on measured optic"""
+        if self.optic_measured is not None:
+            # Perfom ray trace and intersection
+            ray_trace = self._ray_trace_scene(self.optic_measured)
+            ray_pts_meas = rt.plane_intersect(
+                ray_trace, self.params_ray_trace.v_target_center, self.params_ray_trace.v_target_normal
+            )
+
+            # Create image
+            image, xv, yv = rt.histogram_image(
+                bin_res=self.options_ray_trace_vis.hist_bin_res,
+                extent=self.options_ray_trace_vis.hist_extent,
+                pts=ray_pts_meas,
+            )
+
+            # Create ensquared energy curve
+            ee, ws = rt.ensquared_energy(ray_pts_meas, self.options_ray_trace_vis.ensquared_energy_max_semi_width)
+
+            # Save
+            self._ray_trace_output_measured = _RayTraceOutput(ray_trace, image, xv, yv, ee, ws)
+        else:
+            lt.info('No measured optic; skipping measured optic ray trace.')
+
+    def _perform_ray_trace_optic_reference(self):
+        if self.optic_reference is not None:
+            # Perform ray trace and intersection
+            ray_trace = self._ray_trace_scene(self.optic_reference)
+            ray_pts = rt.plane_intersect(
+                ray_trace, self.params_ray_trace.v_target_center, self.params_ray_trace.v_target_normal
+            )
+
+            # Create image
+            image, xv, yv = rt.histogram_image(
+                bin_res=self.options_ray_trace_vis.hist_bin_res,
+                extent=self.options_ray_trace_vis.hist_extent,
+                pts=ray_pts,
+            )
+
+            # Create ensquared energy curve
+            ee, ws = rt.ensquared_energy(ray_pts, self.options_ray_trace_vis.ensquared_energy_max_semi_width)
+
+            # Save data
+            self._ray_trace_output_reference = _RayTraceOutput(ray_trace, image, xv, yv, ee, ws)
+        else:
+            lt.info('No reference optic; skipping reference optic ray trace.')
+
+    def _ray_trace_scene(self, obj: RayTraceable) -> rt.RayTrace:
+        """Performs a raytrace of a simple scene with a source and an optic.
```

Comment:
Same as above. I'm fine with leaving these as doc-strings, but don't want to give the impression that this is a user-facing function.